packages/mobile/src/transactions/feed/queryHelper.ts

```diff
@@ -0,0 +1,269 @@
+import { useTranslation } from 'react-i18next'
+import { useDispatch, useSelector } from 'react-redux'
+import { showError, showMessage } from 'src/alert/actions'
+import { ErrorMessages } from 'src/app/ErrorMessages'
+import config from 'src/geth/networkConfig'
+import useInterval from 'src/hooks/useInterval'
+import { getLocalCurrencyCode } from 'src/localCurrency/selectors'
+import { updateTransactions } from 'src/transactions/actions'
+import { TokenTransaction } from 'src/transactions/types'
+import Logger from 'src/utils/Logger'
+import { walletAddressSelector } from 'src/web3/selectors'
+
+export interface QueryHookResult {
+  loading: boolean
+  error: Error | undefined
+  transactions: TokenTransaction[]
+  fetchingMoreTransactions: boolean
+  fetchMoreTransactions: () => void
+}
+interface PageInfo {
+  startCursor: string
+  endCursor: string
+  hasNextPage: boolean
+  hasPreviousPage: boolean
+}
+export interface QueryResponse {
+  data: {
+    tokenTransactionsV2: {
+      transactions: TokenTransaction[]
+      pageInfo: PageInfo
+    }
+  }
+}
+
+const TAG = 'transactions/feed/queryHelper'
+
+// Query poll interval
+const POLL_INTERVAL = 10000 // 10 secs
+
+function addDeduplicatedTransactions(
+  fetchedTransactions: TokenTransaction[],
+  setFetchedTransactions: (txs: TokenTransaction[]) => void
+) {
+  return (transactions: TokenTransaction[]) => {
+    const currentHashes = new Set(fetchedTransactions.map((tx) => tx.transactionHash))
+
+    const transactionsWithoutDuplicatedHash = fetchedTransactions.concat(
+      transactions.filter((tx) => !currentHashes.has(tx.transactionHash))
+    )
+
+    transactionsWithoutDuplicatedHash.sort((a, b) => {
+      return b.timestamp - a.timestamp
+    })
+
+    setFetchedTransactions(transactionsWithoutDuplicatedHash)
+  }
+}
+
+export function useFetchTransactions(): QueryHookResult {
+  const { t } = useTranslation()
+  const dispatch = useDispatch()
+  const address = useSelector(walletAddressSelector)
+  const localCurrencyCode = useSelector(getLocalCurrencyCode)
+
+  const [pageInfo, setPageInfo] = useState<PageInfo | null>(null)
+  const [fetchingMoreTransactions, setFetchingMoreTransactions] = useState(false)
+  const [transactions, setTransactions] = useState<TokenTransaction[]>([])
+
+  // Update the counter variable every |POLL_INTERVAL| so that a query is made to the backend.
+  const [counter, setCounter] = useState(0)
+  useInterval(() => setCounter((n) => n + 1), POLL_INTERVAL)
+
+  const addTransactions = addDeduplicatedTransactions(transactions, setTransactions)
+
+  const handleResult = (result: QueryResponse, paginatedResult: boolean) => {
+    Logger.info(TAG, `Fetched ${paginatedResult ? 'next page' : 'new'} transactions`)
+
+    const returnedTransactions = result.data?.tokenTransactionsV2?.transactions
+    if (returnedTransactions?.length) {
+      addTransactions(returnedTransactions)
+      // We store non-paginated results in redux
```

Comment:
i've been on holiday too long but why do we have non paginated results again? maybe this info can be added to the comment? :)

Reply:
We are using those results to show the users the list of transactions without making any request.